moc_openshift.py

```diff
@@ -29,24 +28,13 @@ def create_rolebindings(self, project_name, user_name, role):
     def update_rolebindings(self, project_name, role, rolebindings_json):
         return
 
-    @abc.abstractmethod
-    def get_configmap_data(self, configmap) -> dict():
-        return dict()
-
-    def set_config_settings(self):
-        config = self.get_configmap_data("openshift-acct-mgt-config")
-        self.id_provider = config["identityProvider"]
-        self.logger.info(f"identity provider set to {self.id_provider}")
-
-    def set_namespace(self, namespace):
-        self.namespace = namespace
+    def get_identity_provider(self):
+        return os.environ["IDENTITY_PROVIDER"]
```

Comment:
It's a good practice use some sort of namespace prefixing on your environment variables (e.g., `ACCT_MGT_IDENTITY_PROVIDER` instead of just `IDENTITY_PROVIDER`). This avoids conflicts when multiple services need similar settings (e.g., if you have two containers that both want an environment variable named `PASSWORD` you can't put all of your environment variables in a single configmap).

Using a prefix like this also makes it easier to inspect your variables, because you can do something like `env |grep MYPREFIX` to only see the variables in which you are interested.

Reply:
OK, I'll make that change.

When I put a prefix on things I generally mean that it is a special case.  For example, when I need to make a distinction between a moc quota definition and an openshift quota definition vs any other quota.  In this case, since the application was not sharing the environment variables with any other application, I didn't feel the need to specify in the configmap nor the environment that it is an identity provider that is somehow special from other identity providers.